tests/test_e2e_15_maintenance.py

```diff
@@ -815,7 +816,79 @@ def test_080_delete_non_existent_mw_on_switch_should_fail(self):
         delete_response = requests.delete(api_url)
         assert delete_response.status_code == 404
 
-    def test_085_end_non_existent_running_mw_on_switch_should_fail(self):
+    def test_085_end_running_mw_on_switch(self):
+        self.restart_and_create_circuit()
+
+        # Sets up the maintenance window information
+        mw_start_delay = 30
+        mw_duration = 60
+        start = datetime.now() + timedelta(seconds=mw_start_delay)
+        end = start + timedelta(seconds=mw_duration)
+
+        # Sets up the maintenance window data
+        payload = {
+            "description": "mw for test 85",
+            "start": start.strftime(TIME_FMT),
+            "end": end.strftime(TIME_FMT),
+            "items": [
+                "00:00:00:00:00:00:00:02"
+            ]
+        }
+
+        # Creates a new maintenance window
+        api_url = KYTOS_API + '/maintenance'
+        response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
+        data = response.json()
+
+        # Extracts the maintenance window id from the JSON structure
+        mw_id = data["mw_id"]
+
+        # Gets the maintenance schema
+        api_url = KYTOS_API + '/maintenance/' + mw_id
+        response = requests.get(api_url)
+        assert response.status_code == 200
+        json_data = response.json()
+        assert json_data['id'] == mw_id
+
+        # Waits for the MW to start
+        time.sleep(mw_start_delay + 5)
+
+        # Verifies the flow behavior during the maintenance
+        s2 = self.net.net.get('s2')
+        flows_s2 = s2.dpctl('dump-flows')
+        assert 'dl_vlan=100' not in flows_s2
+        assert len(flows_s2.split('\r\n ')) == 1
+
+        # Checks connectivity during maintenance
+        h11, h3 = self.net.net.get('h11', 'h3')
+        h11.cmd('ip link add link %s name vlan100 type vlan id 100' % (h11.intfNames()[0]))
+        h11.cmd('ip link set up vlan100')
+        h11.cmd('ip addr add 100.0.0.11/24 dev vlan100')
+        h3.cmd('ip link add link %s name vlan100 type vlan id 100' % (h3.intfNames()[0]))
+        h3.cmd('ip link set up vlan100')
+        h3.cmd('ip addr add 100.0.0.2/24 dev vlan100')
+        result = h11.cmd('ping -c1 100.0.0.2')
+        assert ', 0% packet loss,' in result
+
+        # Ends the maintenance window information
+        api_url = KYTOS_API + '/maintenance/' + mw_id + '/end'
+        end_response = requests.patch(api_url)
+        assert end_response.status_code == 201
+
+        # Waits to the time that the MW should be ended but instead will be running (extended)
+        time.sleep(10)
```

Comment:
```suggestion
        time.sleep(3)
```

I would say that 10 seconds is too much here because when we request to end the MW immediately, the wait time should be only a couple of seconds to allow Kytos to restore the path.